### PR TITLE
Update to react-bootstrap-table  version v2.3.1

### PR DESCRIPTION
**Pull request by scharf**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

I used the [`react-bootstrap-table` documentation](http://allenfang.github.io/react-bootstrap-table/docs.html) and converted it to typescript.

### CODE REVIEW

**Comment by scharf** [comment]
I checked the code again and amended the commit. I also made sure that the typescript code is has the same layout as the documentation. This will make it easier in the future to keep in sync with the documented API

**Comment by scharf** [comment]
pushed version to `v2.3.1`

**Comment by vvakame** [comment]
We can't get response from author.
please ping me again in 2 days. I'll merge this PR.

**Comment by flaub** [comment]
I've mixed feelings about using the latest typescript features.

Otherwise, LGTM

File: react-bootstrap-table/react-bootstrap-table.d.ts [hunk 1]
```diff
@@ -520,11 +558,13 @@ declare module "react-bootstrap-table" {
 		options?: any;
 	}
 	
+	export type FilterType = 'TextFilter' | 'RegexFilter' | 'SelectFilter' | 'NumberFilter' | 'DateFilter' | 'CustomFilter';
```
Comment (flaub) [inline react-bootstrap-table/react-bootstrap-table.d.ts]: How can users add their custom filters? Is there a way to make this type extensible?

Comment (scharf) [inline react-bootstrap-table/react-bootstrap-table.d.ts]: As I understand the code, `'CustomFilter'` is the way to add a [custom filter](https://github.com/AllenFang/react-bootstrap-table/blob/80bad67aa31a1471e3ffa06b9a98baa4382c086f/src/Const.js#L28-L35).

Hmm, but maybe the `FilterType` should be a `string`. The documentation seems to suggest that any string is allowed here.

**Comment by scharf** [comment]
@flaub, I agree with the mixed feelings about [string literal types](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#string-literal-types). Especially, because they cause parsing problems with older versions of typescript....

File: react-bootstrap-table/react-bootstrap-table.d.ts (hunk 1 above)
Comment (scharf) [inline react-bootstrap-table/react-bootstrap-table.d.ts]: OTOH, the nice thing about declaring `FilterType` is that a good editor/IDE gives you code completion. If someone wants to use a custom filter, they could use

```typescript
{
     filter: 'MyCustomFilter' as FilterType
}
```

Which would keep the type checking and the code completion while making it explicit that you are using a custom filter...

Comment (flaub) [inline react-bootstrap-table/react-bootstrap-table.d.ts]: Ah thanks. I agree with your interpretation of the code. Also good idea on how to extend the type.

**Comment by vvakame** [comment]
It merged. this repository support latest stable tsc only. it is 1.8.10 now.

**Pull request merged**